pkg/apis/core/validation/validation.go

```diff
@@ -6094,3 +6145,177 @@ func ValidateSpreadConstraintNotRepeat(fldPath *field.Path, constraint core.Topo
+		}
+	}
+
+	// IPFamilyPolicy stand alone validation
+	if service.Spec.IPFamilyPolicy != nil /* nil is ok, defaulted in alloc */ {
+		// must have a supported value
+		if !supportedServiceIPFamilyPolicy.Has(string(*(service.Spec.IPFamilyPolicy))) {
+			allErrs = append(allErrs, field.NotSupported(ipFamilyPolicyField, service.Spec.IPFamilyPolicy, supportedServiceIPFamilyPolicy.List()))
+		}
+
+		// gate must be on
+		if !isHeadlessNoSelectorSvc(service) {
+			if !utilfeature.DefaultFeatureGate.Enabled(features.IPv6DualStack) && *(service.Spec.IPFamilyPolicy) == core.RequireDualStack {
+				allErrs = append(allErrs, field.Invalid(ipFamilyPolicyField, service.Spec.IPFamilyPolicy, "RequireDualStack are only allowed when IPv6DualStack feature gate is enabled"))
+			}
+		}
+	}
+
+	// clusterIPs stand alone validation
+	// valid ips with None and empty string handling
+	// no duplication
+	seen = sets.String{}
+	for i, clusterIP := range service.Spec.ClusterIPs {
+		// duplicate check
+		if seen.Has(clusterIP) {
+			allErrs = append(allErrs, field.Duplicate(clusterIPsField.Index(i), clusterIP))
+		}
+		seen.Insert(string(clusterIP))
+
+		// valid at first location only. if and only if len(clusterIPs) == 1
+		if i == 0 && clusterIP == "" {
+			if len(service.Spec.ClusterIPs) > 1 {
+				allErrs = append(allErrs, field.Invalid(clusterIPsField, service.Spec.ClusterIPs, "empty string entry in ClusterIPs is only allowed for services with single ClusterIP"))
+				hasMisplacedNoneEmptyString = true
+			}
+			continue
+		}
+
+		// valid at first location only. if and only if len(clusterIPs) == 1
+		if i == 0 && clusterIP == core.ClusterIPNone {
+			if len(service.Spec.ClusterIPs) > 1 {
+				hasMisplacedNoneEmptyString = true
+				allErrs = append(allErrs, field.Invalid(clusterIPsField, service.Spec.ClusterIPs, "'None' entry in ClusterIPs is only allowed for services with single ClusterIP"))
+			}
+			continue
+		}
+		// we purposfully use different error messages (not IP validation error to help user understand what is wrong)
+		if i != 0 && clusterIP == core.ClusterIPNone || clusterIP == "" {
+			hasMisplacedNoneEmptyString = true
+			allErrs = append(allErrs, field.Invalid(clusterIPsField.Index(i), service.Spec.ClusterIPs, "'None' and empty string is only allowed as the first entry in clusterIPs"))
+			continue
+		}
+
+		// is it valid ip?
+		errorMessages := validation.IsValidIP(clusterIP)
+		hasInvalidIPs = (len(errorMessages) != 0) || hasInvalidIPs
+		for _, msg := range errorMessages {
+			allErrs = append(allErrs, field.Invalid(clusterIPsField.Index(i), clusterIP, msg))
+		}
+	}
+
+	// max two
+	if len(service.Spec.ClusterIPs) > 2 {
+		allErrs = append(allErrs, field.Invalid(specPath.Child("clusterIPs"), service.Spec.ClusterIPs, "maximum allowed entries in clusterIPs is two"))
+	}
+
+	// dual stack feature gate validation
+	if !utilfeature.DefaultFeatureGate.Enabled(features.IPv6DualStack) && len(service.Spec.ClusterIPs) > 1 {
```

Comment:
ditto above re gate

Reply:
not sure i understand. please elaborate

Reply:
I'm going to resolve this - see above :)